core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1729,6 +1729,46 @@ public void testNoTrailingLocationSlash() {
         meta.location());
   }
 
+  @Test
+  public void testCompressionDefaultPropertyForParquet() {
+    Map<String, String> properties =
+        ImmutableMap.of(TableProperties.DEFAULT_FILE_FORMAT, FileFormat.PARQUET.name());
+    TableMetadata metadata =
+        TableMetadata.newTableMetadata(
+            TEST_SCHEMA, SPEC_5, SORT_ORDER_3, "/some-location/", properties);
+
+    Assertions.assertThat(metadata.property(TableProperties.PARQUET_COMPRESSION, null))
+        .withFailMessage(
```

Comment:
I don't think we should be using `withFailMessage()` because it will hide all contextual information and will only show this message. We rather want to use `.as()` in all of these cases here

Reply:
Read the docs, I see the difference between the two now :)